qiskit_ibm/ibm_provider.py

```diff
@@ -0,0 +1,965 @@
+        cls._providers = OrderedDict()
+
+    @classmethod
+    def _get_provider(
+            cls,
+            token: Optional[str] = None,
+            url: Optional[str] = None,
+            hub: Optional[str] = None,
+            group: Optional[str] = None,
+            project: Optional[str] = None,
+    ) -> 'IBMProvider':
+        """Return a provider for a single hub/group/project combination.
+
+        Args:
+            token: IBM Quantum token.
+            url: URL for the IBM Quantum authentication server.
+            hub: Name of the hub.
+            group: Name of the group.
+            project: Name of the project.
+
+        Returns:
+            A provider that matches the specified criteria or default provider.
+
+        Raises:
+            IBMProviderError: If no provider matches the specified criteria,
+                or more than one provider matches the specified criteria.
+        """
+        providers = cls.providers(token=token, url=url)
+        # Prevent edge case where no hubs are available.
+        if not providers:
+            logger.warning('No Hub/Group/Projects could be found for this '
+                           'account.')
+            return None
+        # The provider for the default open access project.
+        default_provider = providers[0]
+        # If any `hub`, `group`, or `project` is specified, return the corresponding provider.
+        if any([hub, group, project]):
+            providers = cls.providers(token=token, url=url, hub=hub, group=group, project=project)
+            if not providers:
+                raise IBMProviderError('No provider matches the specified criteria: '
+                                       'hub = {}, group = {}, project = {}'
+                                       .format(hub, group, project))
+            if len(providers) > 1:
+                raise IBMProviderError('More than one provider matches the specified criteria.'
+                                       'hub = {}, group = {}, project = {}'
+                                       .format(hub, group, project))
+            default_provider = providers[0]
+        return default_provider
+
+    def __init__(
+            self,
+            token: Optional[str] = None,
+            url: Optional[str] = None,
+            hub: Optional[str] = None,
+            group: Optional[str] = None,
+            project: Optional[str] = None,
+            **kwargs: Any
+    ) -> None:
+        # pylint: disable=unused-argument,unsubscriptable-object
+        super().__init__()
+        account = self._account
+        if account:
+            self.credentials = self._construct_provider_credentials(
+                account['credentials'],
+                account['preferences'],
+                account['auth_client'],
+                account['service_urls'],
+                hub,
+                group,
+                project
+            )
```

Comment:
If `account` is to be turned into a `Credentials`, why can't it just be a `Credentials` to begin with?

Reply:
Fixed!